src/integration-tests/take-1inch.test.ts

```diff
@@ -0,0 +1,406 @@
+import { AjnaSDK, FungiblePool, Provider } from '@ajna-finance/sdk';
+import axios from 'axios';
+import { expect } from 'chai';
+import { BigNumber, Contract, ethers, utils, Wallet } from 'ethers';
+import sinon from 'sinon';
+import { AjnaKeeperTaker__factory } from '../../typechain-types/factories/contracts';
+import * as oneInch from '../1inch';
+import ERC20_ABI from '../abis/erc20.abi.json';
+import { configureAjna, LiquiditySource } from '../config-types';
+import { SECONDS_PER_DAY } from '../constants';
+import { getLoansToKick, kick } from '../kick';
+import { getLiquidationsToTake, takeLiquidation } from '../take';
+import { arrayFromAsync, decimaledToWei } from '../utils';
+import { depositQuoteToken, drawDebt } from './loan-helpers';
+import './subgraph-mock';
+import {
+  makeGetHighestMeaningfulBucket,
+  makeGetLiquidationsFromSdk,
+  makeGetLoansFromSdk,
+  overrideGetHighestMeaningfulBucket,
+  overrideGetLiquidations,
+  overrideGetLoans,
+} from './subgraph-mock';
+import { MAINNET_CONFIG, USER1_MNEMONIC } from './test-config';
+import {
+  getProvider,
+  impersonateSigner,
+  increaseTime,
+  resetHardhat,
+  setBalance,
+} from './test-utils';
+
+describe('Take with 1inch Integration', () => {
+  let provider: Provider;
+  let axiosGetStub: sinon.SinonStub;
+  let pool: FungiblePool;
+  let signer: Wallet;
+  let keeperTakerAddress: string;
+  let borrower: string;
+  let quoteToken: Contract;
+  let collateralToken: Contract;
+
+  const ONE_INCH_QUOTE_RESPONSE = {
+    dstAmount: '1000000000000000000',
+  };
+
+  const ONE_INCH_SWAP_RESPONSE = {
+    tx: {
+      to: '0x1111111254EEB25477B68fb85Ed929f73A960582',
+      data: '0x12345678deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef',
+      value: '0',
+      gas: '200000',
+    },
+  };
+
+  before(async () => {
+    process.env.ONEINCH_API = 'https://api.1inch.io/v6.0';
+    process.env.ONEINCH_API_KEY = 'mock_api_key';
+    provider = getProvider();
+    await resetHardhat();
+  });
+
+  beforeEach(async () => {
+    await resetHardhat();
+    axiosGetStub = sinon.stub(axios, 'get');
+    axiosGetStub
+      .withArgs(sinon.match(/\/quote$/), sinon.match.any)
+      .callsFake(() => Promise.resolve({ data: ONE_INCH_QUOTE_RESPONSE }));
+    axiosGetStub
+      .withArgs(sinon.match(/\/swap$/), sinon.match.any)
+      .callsFake(() => Promise.resolve({ data: ONE_INCH_SWAP_RESPONSE }));
+
+    sinon
```

Comment:
The git commit where this canned data was introduced  has timestamp 1745295561, which maps to mainnet block 22322107.  I tried updating the mainnet fork block to that and restarted the hh node, but it didn't help.  Was this API data targeting mainnet or another chain?  The router paths returned by the API will be chain-specific.

It does seem related to the swap.  If I comment out the `pool.take` call from _AjnaKeeperTaker_, the TX  succeeds, so it isn't an issue with the owner or pool validation.